src/main/java/net/dv8tion/jda/core/requests/restaction/RoleOrderAction.java

```diff
@@ -92,6 +95,11 @@ public Guild getGuild()
     @Override
     protected void finalizeData()
     {
+        final Member self = guild.getSelfMember();
+        final int selfPos = guild.getRoles().indexOf(self.getRoles().get(0));
+        if (!self.hasPermission(Permission.MANAGE_ROLES))
```

Comment:
Check this before the self.getRoles(). If you do, then the member would **have** to have a role to actually have this permission, thuse you don't need to actually check if they have any roles before calling `self.getRoles().get(0)`